src/ExpressionEvaluator/Core/Source/FunctionResolver/CSharp/Scanner.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator
+{
+    internal enum TokenKind
```

Comment:
Are these types needed outside of the Parser? If not it would be better to encapsulate them in the Parser class.